src/components/SubmitEvent.tsx

```diff
@@ -71,8 +67,7 @@ const SubmitEvent = ({
     routeUser(`/evenement/${formData.id}`);
   };
 
-  const handleClick: () => void =
-    updateOrAdd === "add" ? pushEventToArray : updateEventInArray;
+  const handleClick = updateOrAdd === "add" ? pushEventToArray : updateEventInArray;
```

Comment:
Again, don't need to provide inline types.  If you remove the explicit type declaration and hover over `handleClick`, you should see the same type regardless.  Also, this is just my eyes, but typing functions like this feels super hard to read what's going on.  If I ever need to provide a type for a function.  I'll almost always pull out into it's own type and apply that type, because `: () => void =` is unicode soup to me